schema/snapcraft.yaml

```diff
@@ -123,6 +123,11 @@ properties:
             uniqueItems: true
             items:
               type: string
+          aliases:
+            type: array
+            items:
+              type: string
+              pattern: "^[a-zA-Z0-9][-_.a-zA-Z0-9]*$"
```

Comment:
You're using the same regex as that used in snapd, but why are aliases validated differently than apps (e.g. why can aliases end in a hyphen but apps can't)? I've reached out to snapd to see why they're different.